### PR TITLE
Final Pull Request

### PR DESCRIPTION
**Pull request by Peter-Dumbari**
## In this PR:

_I worked according the below instructions and all the requirement were implemented_

- [ ] Create a new Rails app called 'hello-rails-react'.
- [ ] Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- [ ] Verify that the jsbundling-rails gem is added to your Gemfile.
- [ ] Initialize your project with Git.
- [ ] Install needed node dependencies as described in the previous lesson's tutorial.
- [ ] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [ ] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [ ] Make sure that your project has a Postgres database set up.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [ ] Create a static view that will be the root of your app.
- [ ] Create the App component with react-router and render it in your static view.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [ ] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [ ] Display the random greeting in your Greeting component.

### CODE REVIEW

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Peter-Dumbari   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
 - No linter error :heavy_check_mark: 
 - You implemented correct the project's requirements :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,12 @@
...
   # Defines the root path route ("/")
   # root "articles#index"
+
+  root 'root#index'
+
+
+  namespace :api do
+    resources :greetings
+  end
+  get 'greetings/random', to: 'greetings#random_greeting'
```
[the remark this answers is not in the document]
Comment (Peter-Dumbari) [reply config/routes.rb:13]: @levy002 , if I make the `greetings#random` root routes then you can't get access to the React application

File: README.md [hunk 2]
```diff
@@ -1,24 +1,172 @@
...
+- For eslint run the following:
+
+```sh
+  rubocop
+```
+
+### Deployment
+
+- Heroku etc.
+
```
Comment (levy002) [inline README.md]: - [ ] Kindly make sure to remove the `Deployment` section as long as your project is not deployed. KIndly remove this section in order to have a clean and professional Readme.

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Peter-Dumbari   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 
- Your project is working as expected :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @Peter-Dumbari 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Implemented suggested changes
-  ✅ Your README looks good, well done!
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕

- [OPTIONAL] Kindly make sure to follow the correct Gitflow for your upcoming projects. You can learn more about it [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) 📑

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**